js/slider.js

```diff
@@ -0,0 +1,86 @@
+window.onload = () => {
+    nextSlide();
+    slideRotation();
+};
+
+var nextButton = document.querySelector(".slider-button-next");
+nextButton.addEventListener("click", nextSlide);
+
+var prevButton = document.querySelector(".slider-button-prev");
+prevButton.addEventListener("click", prevSlide);
+
+var arrowButtons = document.querySelectorAll(".slider-button");
+arrowButtons.forEach(button => {button.addEventListener("click", cancelSlideRotation)});
+
+var dots = document.querySelectorAll(".slider-dots-element");
+
+var thisAd = 0;
+var slides = [
+  {
+    name: "slide1",
+    url: "url(../fotos/cat_background.jpg)"
+  },
+  {
+    name: "slide2",
+    url: "url(../fotos/cat_background2.jpg)"
+  },
+  {
+    name: "slide3",
+    url: "url(../fotos/cat_background3.jpg)"
+  }
+];
+
+var rotate;
+function slideRotation() {
+    rotate = setInterval(nextSlide, 7000);
```

Comment:
Jak się szybko kliknie w przyciski next i prev to transition slajdów jest raz płynna, raz urywana

Reply:
transition ustawione jest na wartość 1s. jeśli klikasz szybko (wartość mniejsza niż jedna sek.), to css przerywasz animację. CSS nie kolejkuje animacji - nie czeka aż jedna się skończy i dopiero puszcza następną. Robi to od razu. Zmieniłem wartość na 0.5s żeby było to mniej zauważalne.